web/server/api/endpoints/files.py

```diff
@@ -72,72 +148,32 @@ def walk_path(
                     )
         return sorted(directories, key=lambda x: x.name), sorted(files, key=lambda x: x.name)
 
-    directories, files = walk_path(settings.project_path)
+    directories, files = walk_path(path)
+    relative_path = str(Path(path).relative_to(settings.project_path))
     return models.Directory(
-        name=os.path.basename(settings.project_path),
-        path="",
+        name=os.path.basename(path),
+        path="" if relative_path == "." else relative_path,
         directories=directories,
         files=files,
     )
 
 
-@router.get("/{path:path}", response_model=models.File)
-def get_file(
-    path: str = Depends(validate_path),
-    settings: Settings = Depends(get_settings),
-    path_mapping: t.Dict[Path, models.FileType] = Depends(get_path_mapping),
+def _get_file_with_content(
+    path: Path,
+    settings: Settings,
+    path_mapping: t.Dict[Path, models.FileType],
 ) -> models.File:
     """Get a file, including its contents."""
+    file_path = settings.project_path / path
+
     try:
-        with open(settings.project_path / path) as f:
+        with open(file_path) as f:
             content = f.read()
-    except FileNotFoundError:
-        raise HTTPException(status_code=HTTP_404_NOT_FOUND)
+    except FileNotFoundError as e:
+        raise e
     return models.File(
-        name=os.path.basename(path), path=path, content=content, type=path_mapping.get(Path(path))
-    )
-
-
-@router.post("/{path:path}", response_model=models.File)
-async def write_file(
-    content: str = Body("", embed=True),
-    new_path: t.Optional[str] = Body(None, embed=True),
-    path: str = Depends(validate_path),
-    settings: Settings = Depends(get_settings),
-    context: Context = Depends(get_context),
-    path_mapping: t.Dict[Path, models.FileType] = Depends(get_path_mapping),
-) -> models.File:
-    """Create, update, or rename a file."""
-    path_or_new_path = path
-    if new_path:
-        path_or_new_path = validate_path(new_path, context)
-        replace_file(settings.project_path / path, settings.project_path / path_or_new_path)
-    else:
-        (settings.project_path / path_or_new_path).write_text(content, encoding="utf-8")
-
-    content = (settings.project_path / path_or_new_path).read_text()
-    return models.File(
-        name=os.path.basename(path_or_new_path),
-        path=path_or_new_path,
+        name=os.path.basename(path),
```

Comment:
I think the following should work since `path` is a Path object and not a string.
```suggestion
        name=path.name,
```